analyzers/tests/SonarAnalyzer.CSharp.Styling.Test/TestCases/LambdaParameterName.cs

```diff
@@ -72,25 +73,52 @@ public void Initialize()
     protected void RegisterSonarWhateverReportingContext(Action<SonarWhateverReportingContext> action) { }
-    protected void RegisterSonarSometing(Action<SonarSometing> action) { }
-    protected void RegisterSometingAnalysisContext(Action<SometingAnalysisContext> action) { }
-    protected void RegisterSometingReportingContext(Action<SometingReportingContext> action) { }
-    protected void RegisterSonarSometingContext(Action<SonarSometingContext> action) { }
+    protected void RegisterSonarSomething(Action<SonarSomething> action) { }
+    protected void RegisterSomethingAnalysisContext(Action<SomethingAnalysisContext> action) { }
+    protected void RegisterSomethingReportingContext(Action<SomethingReportingContext> action) { }
+    protected void RegisterSonarSomethingContext(Action<SonarSomethingContext> action) { }
 
     // Well-known expected classes patterns
     public class SonarWhateverAnalysisContext { }
     public class SonarWhateverReportingContext { }
     // Unexpected types
-    public class SonarSometing { }
-    public class SometingAnalysisContext { }
-    public class SometingReportingContext { }
-    public class SonarSometingContext { }
+    public class SonarSomething { }
+    public class SomethingAnalysisContext { }
+    public class SomethingReportingContext { }
+    public class SonarSomethingContext { }
+}
+
+public class CustomDelegates
+{
+    public delegate void ParameterNamedI(int i);
+    public delegate void ParameterNamedTest(int test);
+    public delegate void ParameterNamedCamelCasing(int camelCasing);
+
+    public void Test()
+    {
+        ParameterNamedI delegate1 = i => { }; // Compliant "i" matches the parameter name of the delegate
+        ParameterNamedI delegate2 = j => { }; // Noncompliant
+        ParameterNamedI delegate3 = x => { }; // Compliant
+
+        ParameterNamedTest delegate4 = test => { };     // Compliant
+        ParameterNamedTest delegate5 = someTest => { }; // Noncompliant
+        ParameterNamedTest delegate6 = testSome => { }; // Noncompliant
+
+        ParameterNamedCamelCasing delegate7 = camelCasing => { }; // Compliant
+        ParameterNamedCamelCasing delegate8 = camelcasing => { }; // Noncompliant
+        ParameterNamedCamelCasing delegate9 = camel => { };     // Noncompliant
+        ParameterNamedCamelCasing delegate10 = casing => { };    // Noncompliant
+
+        Func<int, int> function = arg => 0;  // Noncompliant, the delegate parameter is named "arg" (https://learn.microsoft.com/en-us/dotnet/api/system.func-2) but we do not allow that for Func<T, TResult>
+        Action<int> action = obj => { };     // Noncompliant, the delegate parameter is named "obj" (https://learn.microsoft.com/en-us/dotnet/api/system.action-1) but we do not allow that for Action<T>
+        new List<int>().Exists(obj => true); // Compliant. List.Exists uses System.Predicate<T> instead of System.Func<T, bool>
```

Comment:
I think is an FN.

I don't know how often delegates are used in Collection type (and other types).
It looks to me, from the point of view of the coding style, we should use `x`.

I guess, it is an FN only when matching the delegate parameter name, so maybe it is fine.

Reply:
Yes. This is very niche. `System.Predicate` is used by a couple of framework methods only. It predates `Action` and `Func` and is considered a legacy today. We could add it to `IsFuncOrAction`, but I do not think it is worth it. I just wanted to document it.